perf/proxy_service_notifier.go

```diff
@@ -0,0 +1,128 @@
+package perf
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"github.com/evergreen-ci/cedar/model"
+	"github.com/evergreen-ci/utility"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+// ProxyService is the interface for the Proxy Service.
+type ProxyService interface {
+	ReportNewPerformanceDataAvailability(context.Context, model.PerformanceTestResultId) error
+}
+
+type proxyServiceClient struct {
+	user    string
+	token   string
+	baseURL string
+}
+
+// NewProxyService creates a new ProxyService.
+func NewProxyService(options model.ProxyServiceOptions) ProxyService {
+	return &proxyServiceClient{user: options.User, token: options.Token, baseURL: options.BaseURL}
+}
+
+// ReportNewPerformanceDataAvailability takes a PerformanceTestResultId and tries to report its data to a ProxyService.
+func (spc *proxyServiceClient) ReportNewPerformanceDataAvailability(ctx context.Context, data model.PerformanceTestResultId) error {
+	startAt := time.Now()
+
+	if err := spc.doRequest(http.MethodPost, spc.baseURL+"/rabbitmq/performance_data_update/patch", ctx, data); err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
+			"message": "Failed to report new time series performance data availability to proxy service",
+			"update":  data,
+		}))
+		return nil
+	}
+
+	grip.Debug(message.Fields{
+		"message":       "Reported new time series performance data availability to the proxy service",
+		"update":        data,
+		"duration_secs": time.Since(startAt).Seconds(),
+	})
+
+	return nil
+}
+
+func (spc *proxyServiceClient) doRequest(method, route string, ctx context.Context, in interface{}) error {
```

Comment:
You never actually return errors in this func, you do log them though. I think the logging is redundant since you log the error message on L40 above.